client/src/views/TableDetail.vue

```diff
@@ -88,34 +88,38 @@
     </v-content>
   </v-container>
 </template>
-<script>
-import api from '@/api'
 
-export default {
+<script lang="ts">
+import Vue from 'vue';
+
+import api from '@/api';
+import { KeyValue, TableRow } from '@/types';
+
+export default Vue.extend({
   name: 'TableDetail',
   props: ['workspace', 'table'],
-  data () {
+  data() {
     return {
-      rowKeys:[],
-      headers:[],
+      rowKeys: [] as KeyValue[][],
+      headers: [] as Array<keyof TableRow>,
       tables: [],
       editing: false,
-    }
+    };
   },
   methods: {
-    rowClassName(index) {
-      return index % 2 == 0 ? 'even-row' : 'odd-row';
+    rowClassName(index: number): 'even-row' | 'odd-row' {
```

Comment:
Yes, this sort of thing standard practice, both here and in Python3 (which provides a more explicit syntax for creating a type from a finite set of strings).

This would feel less like overkill if we defined this as a type, and then enforced the use of that type for a variable somewhere else holding a classname for use on some element. In that context, being overly judicious here would prevent errors in the usage of that class name fairly robustly in the future.

I agree that in this case we would be ok using `string` as the type, but I also think in general that imbuing types with as much information as possible is usually a good idea.

Reply:
Ok, I'm convinced :)